articles/django-app-using-postgresql-database/index.md

```diff
@@ -0,0 +1,217 @@
+### Creating a Django Hello World App Using PostgreSQL Database
+> Django is a high-level Python Web framework that encourages rapid development and clean, pragmatic design. Built by experienced developers, it takes care of much of the hassle of Web development, so you can focus on writing your app without needing to reinvent the wheel. It’s free and open-source <cite>[Django](https://www.djangoproject.com)<cite>
+
+> PostgreSQL is a powerful, open-source object-relational database system that uses and extends the SQL language combined with many features that safely store and scale the most complicated data workloads. <cite>[PostgreSQL](https://www.postgresql.org/about/)<cite>
+
+### Prerequisites
+To follow this tutorial you need to have
+ - [Python3](https://www.python.org/downloads/)
+ - [PostgreSQL](https://www.postgresql.org/download/windows/)
+ - [virtualenv](https://pypi.org/project/virtualenv/)
+
+### Step 1 - Create a database and database user we will be creating our database through the command line.
+ ![access psql](/engineering-education/django-app-using-postgresql-database/access_psql.jpg)
+
+Click on the box area
+Click on Enter to accept a default value. input the password that you provide during PostgresSQL installation.
+
+```bash 
+ CREATE DATABASE helloworld;
+ CREATE USER <yourname> WITH PASSWORD '<password>';
+```
+Modifying a few connections parameters for the new database user. 
+```bash 
+ALTER ROLE <yourname> SET client_encoding TO 'utf8';
+ALTER ROLE <yourname> SET default_transaction_isolation TO 'read committed';
+ALTER ROLE <yourname> SET timezone TO 'UTC';
+```
+Next, we have to grant our new user access to the database we created earlier.
+```bash 
+
+ GRANT ALL PRIVILEGES ON DATABASE helloworld TO <yourname>;
+```
+
+### Step - 2 creating a virtual environment using and installing necessary python package
+
+Let's start by creating a folder for our project
+```bash 
+
+ mkdir myproject 
+ cd myproject
+```
+Let's create a virtual environment and activate it.
+```bash 
+virtualenv env
+source venv/bin/activate
+```
+ Installing django and psycopg2
+ ```bash
+ pip install django psycopg2
+ ```
+  - psycopg2 is a popular PostgreSQL database adapter for python programming language.
+  
+Let's create our Hello World Django project
+```bash
+ django-admin startproject django_app
+```
+
+Change your current directory into our Django project django_app
+
+```bash
+cd hello_world
+```
+Let's create an app in our django_app project
+> A Django application is a Python package that is specifically intended for use in a Django project. Django apps are reusable in different Django project. [Django](https://www.djangoproject.com)
+```bash 
+django-admin startapp hello_world
+```
+We need to plug our hello_world app to our django_app to do this navigate to
+django_app/settings.py
+
+```python 
+...
+INSTALLED_APPS = [
+    ...
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'hello_world',
+]
+```
```

Comment:
Instruct readers to add the app name in Installed apps. It will be unclear if you just provide a snippet.